### PR TITLE
feat(member-invites): Send emails on join / invite requests

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/templates/sentry/emails/organization-invite-request.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% extends "sentry/emails/base.html" %}
+
+{% load i18n %}
+
+{% block main %}
+  <h3>Request for Access</h3>
+
+  <p><strong>{{ inviter.get_salutation_name }}</strong> has request for <strong>{{ email }}</strong> to join the <a href="{{ organization_link }}">{{ organization.name }}</a> organization.</p>
```
Comment (maheskett) [inline src/sentry/templates/sentry/emails/organization-invite-request.html]: how about `has requested to invite [email] to the [name] organization`

File: src/sentry/templates/sentry/emails/organization-invite-request.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Request for Access
+
+{{ inviter.get_salutation_name }} has request for {{ email }} to join the {{ organization.name }} organization.
```
Comment (maheskett) [inline src/sentry/templates/sentry/emails/organization-invite-request.txt]: same

File: src/sentry/templates/sentry/emails/organization-invite-request.html (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/templates/sentry/emails/organization-invite-request.html]: Yeah sounds good

File: src/sentry/models/organizationmember.py [hunk 3]
```diff
@@ -235,6 +235,54 @@ def send_invite_email(self):
...
+        if self.requested_to_join:
+            email_args = {
+                "subject": "%s has requested to join %s on Sentry"
+                % (self.email, self.organization.name),
+                "template": "sentry/emails/organization-join-request.txt",
+                "html_template": "sentry/emails/organization-join-request.html",
+            }
+        elif self.requested_to_be_invited:
+            email_args = {
+                "subject": "%s has requested to invite a member to %s on Sentry"
```
Comment (maheskett) [inline src/sentry/models/organizationmember.py]: maybe these subject lines could be shorter and less specific like `Sentry Access Request`

File: tests/sentry/models/test_organizationmember.py [hunk 4]
```diff
@@ -40,6 +40,37 @@ def test_send_invite_email(self):
 
         assert msg.to == ["foo@example.com"]
 
+    def test_send_request_notification_email(self):
```
Comment (maheskett) [inline tests/sentry/models/test_organizationmember.py]: might not hurt to test for the email on the endpoints as well

**Review by maheskett — APPROVED** [review]

File: src/sentry/models/organizationmember.py (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/models/organizationmember.py]: Ahh yeah, good call

File: tests/sentry/models/test_organizationmember.py (hunk 4 above)
Comment (EvanPurkhiser) [reply tests/sentry/models/test_organizationmember.py]: Added

**Pull request merged**